server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala

```diff
@@ -61,6 +61,11 @@ private[deploy] class Worker(
 
   private val WORKER_SHUTDOWN_PRIORITY = 100
   private val shutdown = new AtomicBoolean(false)
+  private val gracefulShutdown = RssConf.workerGracefulShutdown(conf)
+  assert(!gracefulShutdown || (gracefulShutdown &&
+    RssConf.workerRPCPort(conf) != 0 && RssConf.fetchServerPort(conf) != 0 &&
+    RssConf.pushServerPort(conf) != 0 && RssConf.replicateServerPort(conf) != 0),
+    "If enable graceful shutdown, the worker should use stable server port.")
```

Comment:
ping @waitinfuture  How about like this?

Reply:
I think it's good.